src/services/createResolvedRoute.ts

```diff
@@ -0,0 +1,33 @@
+import { ResolvedRoute } from "@/types/resolved"
+import { Route } from "@/types/route"
+import { parseUrl } from "@/services/urlParser"
+import { assembleUrl } from "@/services/urlAssembly"
+import { RouterResolveOptions } from "@/types/RouterResolve"
+import { createResolvedRouteQuery } from "@/services/createResolvedRouteQuery"
+import { getRouteParamValues } from "@/services/paramValidation"
+import { getStateValues } from "@/services/state"
+import { asUrl } from "@/types/url"
+
+export function createResolvedRoute(route: Route, url: string, options?: RouterResolveOptions): ResolvedRoute | undefined
```

Comment:
Why does this accept both a route and a url? I assumed we'd create the url within this method.

Reply:
we decided to duplicate the few lines of createResolvedRoute into getResolvedRouteForUrl so that this doesn't have the overloads